### PR TITLE
Listener refactoring.

### PR DESCRIPTION
**Pull request by fboucquez**
It subscribes using aliases if required.
It allows looking for all the unresolved addresses related to an account.
It allows looking for all the unresolved addresses related to an multisig account like cosigners.
Added lots of integration tests.
Improved ListenerBase

FIxes https://github.com/nemtech/symbol-sdk-java/issues/342

### CODE REVIEW

**Review by rg911 — COMMENTED** [review]
Looks okay to me, There's a new multisigRepo in the listener's constructor, but should be fine creating using the factory. @Wayonb are you okay with it?

The other thing is resolving the alias and multisig by calling the rest. It's completely removed in ts sdk to reduce rest calls, as wallet get banned / connection rejected so often,

File: sdk-core/src/main/java/io/nem/symbol/sdk/infrastructure/ListenerBase.java [hunk 1]
```diff
@@ -280,170 +282,124 @@ private void validateOpen() {
...
-                        accountAddressRestrictionTransaction
-                            .getRestrictionDeletions()
-                            .contains(namespaceId))) {
-              return Observable.just(true);
-            }
-            return Observable.just(false);
-          });
-    }
+  @Override
+  public Observable<Set<UnresolvedAddress>> getAllMultisigAddressesAndAliases(
```
Comment (Wayonb) [inline sdk-core/src/main/java/io/nem/symbol/sdk/infrastructure/ListenerBase.java:343]: Would this be better if implemented in REST?

**Review by Wayonb — APPROVED** [review]
Looks good.  Will play with later.

File: sdk-core/src/main/java/io/nem/symbol/sdk/infrastructure/ListenerBase.java (hunk 1 above)
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/symbol/sdk/infrastructure/ListenerBase.java:343]: We thought (and try) about it, but tbh I would prefer in this case for Rest to just delegate the behavior to the server without any change (at least for now).

**Pull request merged**